lib/Net/SAML2/Binding/SOAP.pm

```diff
@@ -157,7 +158,7 @@ sub request {
         );
     }
 
-    return $self->handle_response($res->decoded_content);
+    return _extract_artifact_message($self->handle_response($res->decoded_content));
```

Comment:
This is a breaking change isn't it?

Reply:
**well that is the question.**  I am not sure we ever properly supported an ArtifactResponse.  There are no current tests and the LogoutResponse definitely did not work.

That being said we could simply do this for the LogoutResponse.  The Response would already work since https://github.com/perl-net-saml2/perl-Net-SAML2/blob/8f284d8bac49ed31113f36012f3b5cb2499b31b7/lib/Net/SAML2/Protocol/Assertion.pm#L116 the assertion looks for an Assertion anywhere in the XML (which is likely not good).

The sub itself returns the original XML if it is not a xpath match for:

```
                        "/samlp:ArtifactResponse/samlp:Response" .
                        "|" .
                        "/samlp:ArtifactResponse/samlp:LogoutResponse
```
So in general Soap messages it has no impact.  In the ArtifactResponse it simply returns the message that the rest of the module can deal with.  I suppose it is possible that someone has a step in their application to remove the LogoutResponse from the ArtifactResponse but the LogoutResponse is the required info (as is the Response in the case of processing the assertion).

Reply:
I think this is breaking, in $client code base we have `$soap_binding->request()` calls and other code looking at the full XML to process the request.

I think request is a somewhat low level call that should be kept as is.

Could we add a new method that does the request and returns the artifact? Eg, `get_artifact($message)` which passes the message to `request` and does the extraction of the message?